social-tw/social-tw-website
# Fix: Default Page Handling in /api/post Route

Please follow the guidelines in [Contribution.md](https://github.com/social-tw/social-tw-website/blob/main/CONTRIBUTING.md) first, then start to create your pull request here.

## Summary

This PR addresses a backend issue in the /api/post route where a missing page query parameter caused a conversion of undefined to NaN, leading to a database query error. The fix implements a check for the page parameter, defaulting it to 1 if not provided, and validating its correctness.

## Linked Issue

fix #213 

## Details

Implemented a check in the /api/post route to ensure the page query parameter is properly handled. The page number defaults to 1 if the parameter is missing and a validation is added to check for a valid numerical page value.

## Impacted Areas

* `packages/relay/src/routes/post.ts`

## Tests

Manual testing was performed to ensure the route correctly handles missing and invalid page numbers.

## Possible Impacts

This change should not affect any existing features other than the intended improvement in the /api/post route handling. No backward compatibility issues are expected.

## Visual Materials

N/A

## Verification Steps

Send a GET request to /api/post without a page query parameter.
Observe the default behavior setting the page to 1.
Send a GET request to /api/post with an invalid page parameter (e.g., non-numeric value).
Observe the error response with a status code of 400.

## Todo

Break down the progress of the PR for everyone to see what else you intend to include in this PR.

-   [x] Implement default and validation for page parameter in /api/post
-   [ ] Unit tests for `function`

## Checklist
